### PR TITLE
Set Gradle user home outside of test project to fix flakiness in NativeServicesIntegrationTest

### PR DESCRIPTION
**Pull request by asodja**
More exactly for `NativeServicesIntegrationTest#native services are initialized with systemProperties` test.

We replace `requireOwnGradleUserHomeDir()` with `withGradleUserHomeDir()` so Gradle user home is located in a different temp dir and not in the test project dir. With that a test project dir is always removed without file lock issues.

### CODE REVIEW

**Comment by asodja** [comment]
@bot-gradle test this

File: subprojects/core/src/integTest/groovy/org/gradle/NativeServicesIntegrationTest.groovy [hunk 1]
```diff
@@ -57,7 +57,7 @@ class NativeServicesIntegrationTest extends AbstractIntegrationSpec {
 
     def "native services are #description with systemProperties == #systemProperties"() {
         given:
-        executer.requireOwnGradleUserHomeDir().withNoExplicitNativeServicesDir()
+        executer.withGradleUserHomeDir(tmpDir.testDirectory).withNoExplicitNativeServicesDir()
```
Comment (wolfs) [inline subprojects/core/src/integTest/groovy/org/gradle/NativeServicesIntegrationTest.groovy:60]: 🤔 My understanding is that this is still in the test project home. Is it not?

Comment (asodja) [reply subprojects/core/src/integTest/groovy/org/gradle/NativeServicesIntegrationTest.groovy:60]: And project dir uses instance defined in:
https://github.com/gradle/gradle/blob/6e60ccd6e76735f829482c3ecd49af5f328789ad/testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractIntegrationSpec.groovy#L76

Comment (wolfs) [reply subprojects/core/src/integTest/groovy/org/gradle/NativeServicesIntegrationTest.groovy:60]: Ah, I see. Can you leave a comment in the test what is going on?

As long as this works I am fine with it.

**Review by wolfs — APPROVED** [review]

File: subprojects/core/src/integTest/groovy/org/gradle/NativeServicesIntegrationTest.groovy (hunk 1 above)
Comment (asodja) [reply subprojects/core/src/integTest/groovy/org/gradle/NativeServicesIntegrationTest.groovy:62]: Good idea, added a comment 👍

**Pull request merged**